beacon-chain/core/validators/validator.go

```diff
@@ -13,11 +13,33 @@ import (
 	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
 	"github.com/prysmaticlabs/prysm/v4/config/params"
 	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
-	mathutil "github.com/prysmaticlabs/prysm/v4/math"
 	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
 	"github.com/prysmaticlabs/prysm/v4/time/slots"
+	"github.com/sirupsen/logrus"
 )
 
+// ValidatorsMaxExitEpochsAndChurn returns the maximum non-FAR_FUTURE_EPOCH exit
+// epoch and the number of them
+func ValidatorsMaxExitEpochsAndChurn(ctx context.Context, s state.BeaconState) (maxExitEpoch primitives.Epoch, churn uint64) {
+	farFutureEpoch := params.BeaconConfig().FarFutureEpoch
+	err := s.ReadFromEveryValidator(func(idx int, val state.ReadOnlyValidator) error {
+		e := val.ExitEpoch()
+		if e != farFutureEpoch {
+			if e > maxExitEpoch {
+				maxExitEpoch = e
+				churn = 1
+			} else if e == maxExitEpoch {
+				churn++
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		logrus.WithError(err).Error("this can't happen")
```

Comment:
Can we remove `this can't happen` or use something else? The description is not very helpful

Reply:
this literally can't happen, it's a quirk of the function signature for `ReadFromEveryValidator` that requires an error to be returned, but this function does never return an error

Reply:
I could change this to an `assert(err == nil)` or a panic for all I know.

Reply:
I personally think either option is better than a `this can't happen` log

Reply:
or silence the error with a comment that says it can't happen
`_ = err`

Reply:
I truly hate linters :) will use this last one